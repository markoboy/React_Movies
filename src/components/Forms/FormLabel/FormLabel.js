import React, { memo } from 'react';
import PropTypes from 'prop-types';

function FormLabel({ label, id }) {
  return label && <label className="form__label" htmlFor={id}>{label}</label>;
}

FormLabel.defaultProps = {
  label: '',
  id: '',
};

FormLabel.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
};

export default memo(FormLabel);
