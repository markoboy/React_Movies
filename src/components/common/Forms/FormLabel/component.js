import React, { memo } from 'react';
import PropTypes from 'prop-types';

function FormLabelComponent({ label, id }) {
  return label && <label className="form__label" htmlFor={id}>{label}</label>;
}

FormLabelComponent.defaultProps = {
  label: '',
  id: '',
};

FormLabelComponent.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
};

export default memo(FormLabelComponent);
