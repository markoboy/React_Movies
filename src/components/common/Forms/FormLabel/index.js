export { default } from './FormLabel.component';
