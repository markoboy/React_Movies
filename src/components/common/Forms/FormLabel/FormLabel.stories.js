import React from 'react';

import FormLabel from './index';

export default {
  component: FormLabel,
  title: 'Common/Forms/FormLabel',
};

const Template = (args) => <FormLabel {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'Form label',
};
