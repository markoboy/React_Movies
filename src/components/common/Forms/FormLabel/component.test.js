import React from 'react';
import { render } from '@testing-library/react';

import FormLabel from './component';

describe('FormLabel Component', () => {
  it('renders with provided label', () => {
    const { container } = render(<FormLabel label="My label" />);

    expect(container).toMatchSnapshot();
  });

  it('doesnt render when label is not provided', () => {
    const { container } = render(<FormLabel />);

    expect(container).toMatchSnapshot();
  });
});
