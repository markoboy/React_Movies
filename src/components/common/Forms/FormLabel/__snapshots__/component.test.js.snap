// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel Component doesnt render when label is not provided 1`] = `
<div>
  
</div>
`;

exports[`FormLabel Component renders with provided label 1`] = `
<div>
  <label
    class="form__label"
    for=""
  >
    My label
  </label>
</div>
`;
